### DISCUSSION TITLE
File managing for mp3 while using player with TFT

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [pschatzmann/arduino-audio-tools#531] [Q&A] [answered]
Hi,Phil,
I'm building a mp3 player with a tft and a sdcard,the two modules are working fine together,
the only thing that makes difficulties is the mp3 managing,i'm using sdfat library to list files and directories,
but i wonder if there are some function managing directly  in the  audio tools library,to list and enum the files to play,
so that i can choose the number of the file and select it in the tft.

**Comment by pschatzmann** [comment, marked as answer]
Did you read the documentation about the AudioPlayer ? https://github.com/pschatzmann/arduino-audio-tools/wiki/The-Audio-Player-Class

**Comment by TangerineDreamer** [comment]
ooops,it seems to be a veritable ali baba cavern!
thanks to point me there :)

**Comment by TangerineDreamer** [comment]
Can i call a mp3 like that?

const char *startFilePath="/DIR2/CD Track 11";
const char* ext="mp3";
AudioSourceSDFAT source(startFilePath, ext);

**Comment by pschatzmann** [comment]
Yes - please note that the path is searched recursively: If you specify "/" it searches all subdirectories.

**Comment by TangerineDreamer** [comment]
I think i made a mistake,last example didn't worked

this one worked : const char startFilePath="/";
const char ext="CD Track 11.mp3";

This one don't work:
const char *startFilePath="DIR2/";
const char* ext="CD Track 11.mp3";
how can i write the repertory?
"DIR2" , "/DIR2/" , "/DIR2" ??

**Comment by pschatzmann** [comment]
I guess this depends on what the file system implementation allows, but I would expect that the syntax /DIR2 should work for all

**Comment by TangerineDreamer** [comment]
No,/DIR2 nor DIR2/ works
It seems your file allocation with sdfat only works with root "/" and don't accept change directories,
maybe we have to use the sdfat - change directories function before,but i'm not good enough for that.
with your SD player file name, it works, eg: audioFile = SD.open("/DIR2/CD Track 11.mp3"); so simple to use ! :)
Is possible to use this with sdfat?

**Comment by pschatzmann** [comment]
The SD library was working fine. I make a small correction for the SDFAT library and it should work with the latest commit.

I tested with 
```
const char *startFilePath="/Bob Dylan";
const char* ext="mp3";
SdSpiConfig sdcfg(PIN_AUDIO_KIT_SD_CARD_CS, DEDICATED_SPI, SD_SCK_MHZ(10) , &AUDIOKIT_SD_SPI);
AudioSourceSDFAT source(startFilePath, ext, sdcfg);
```

**Comment by TangerineDreamer** [comment]
Hey good news!!
it worked at the first time!!
I was a bit afraid because i don't use the audiokit at this time,so i didn't need any sdcfg to fill in the function.
I can go up to 15MHZ,mzybe more/ with the sdcard,same as you.

**Comment by TangerineDreamer** [comment]
the bluetooth communication is a bit hashed,do i level up the transmitter buffer (where?) or the receiver buffer (audiokit)?
